ping/ping6_common.c

```diff
@@ -832,6 +932,71 @@ int ping6_parse_reply(struct ping_rts *rts, socket_st *sock,
 			fflush(stdout);
 			return 0;
 		}
+	} else if (icmph->icmp6_type == ICMPV6_EXT_ECHO_REPLY) {
+		if (!rts->multicast &&
+		    memcmp(&from->sin6_addr.s6_addr, &rts->whereto6.sin6_addr.s6_addr, 16))
+			return 1;
+		if (!is_ours(rts, sock, icmph->icmp6_id))
+			return 1;
+
+		sequence = ntohs(icmph->icmp6_seq);
+		state = icmph->icmp6_seq & 0xe0;
+		printf("Interface: %s\n", rts->interface);
+		switch (icmph->icmp6_code) {
+			case 1:
+				printf("Error: Malformed Query\n");
+				break;
+			case 2:
+				printf("Error: No Such Interface\n");
+				break;
+			case 3:
+				printf("Error: No Such Table Entry\n");
+				break;
+			case 4:
+				printf("Error: Multiple Interfaces Satisfy Query\n");
+				break;
+			default:
+				break;
+		}
+		switch (state) {
+			case 1:
+				printf("State: Incomplete\n");
+				break;
+			case 2:
+				printf("State: Reachable\n");
+				break;
+			case 3:
+				printf("State: Stale\n");
+				break;
+			case 4:
+				printf("State: Delay\n");
+				break;
+			case 5:
+				printf("State: Probe\n");
+				break;
+			case 6:
+				printf("State: Failed\n");
+				break;
+			default:
+				break;
+		}
+		if (icmph->icmp6_code == 0) {
```

Comment:
The returned A/4/6 values are only valid if code=0 and L=1, I recommend checking that here.